intezer_sdk/api.py

```diff
@@ -116,16 +129,114 @@ def request_with_refresh_expired_access_token(self,
                                                   data: dict = None,
                                                   headers: dict = None,
                                                   files: dict = None,
-                                                  stream: bool = None) -> Response:
-        response = self._request(method, path, data, headers, files)
+                                                  stream: bool = None,
+                                                  body: Any = None,
+                                                  base_url: str = None) -> Response:
+        response = self._request(method, path, data, headers, files, stream, body, base_url=base_url)
 
         if response.status_code == HTTPStatus.UNAUTHORIZED:
             self._access_token = None
             self.set_session()
-            response = self._request(method, path, data, headers, files, stream)
+            response = self._request(method, path, data, headers, files, stream, body, base_url)
 
         return response
 
+    def _set_access_token(self, api_key: str):
+        response = requests.post(f'{self.full_url}/get-access-token',
+                                 json={'api_key': api_key},
+                                 verify=self.verify_ssl)
+
+        if response.status_code in (HTTPStatus.UNAUTHORIZED, HTTPStatus.BAD_REQUEST):
+            raise errors.InvalidApiKeyError(response)
+        if response.status_code != HTTPStatus.OK:
+            raise_for_status(response)
+
+        self._access_token = response.json()['result']
+
+    def set_session(self):
+        self._session = requests.session()
+        self._session.mount('https://', requests.adapters.HTTPAdapter(max_retries=3))
+        self._session.verify = self.verify_ssl
+        self._set_access_token(self.api_key)
+        self._session.headers['Authorization'] = 'Bearer {}'.format(self._access_token)
+        self._session.headers['User-Agent'] = self.user_agent
+
+    @staticmethod
+    def _assert_analysis_response_status_code(response: Response):
```

Comment:
all of these methods should move to IntezerApi since they are relevant to specific routes